Package/Default.sublime-keymap

```diff
@@ -84,14 +84,30 @@
             { "key": "setting.packagedev.syntax.input_helpers" },
         ]
     },
-    {   // [optional] auto-insert newline after double-colon (contexts) and 'captures'
+    {   // [optional] auto-insert newline after double-colon of 'captures'
         "keys": [":"], "command": "insert", "args": { "characters": ":\n" }, "context": [
-            { "key": "preceding_text", "operator": "regex_match", "operand": "^ {0,2}[^\\s:]+$|^ {4,}(- *)?(captures)+$", "match_all": true },
+            { "key": "preceding_text", "operator": "regex_match", "operand": "^ {4,}(- *)?(captures)+$", "match_all": true },
             { "key": "selection_empty", "match_all": true },
             { "key": "selector", "operand": "meta.block.contexts.sublime-syntax" },
             { "key": "setting.packagedev.syntax.input_helpers" },
         ]
     },
+    {   // [optional] trim space after colon and insert newline.
+        "keys": ["enter"],  "command": "run_macro_file", "args": {"file": "res://Packages/PackageDev/Package/Sublime Text Syntax Definition/Trim Add Newline.sublime-macro"}, "context": [
+            { "key": "preceding_text", "operator": "regex_match", "operand": "^ {4,}(?:- )?[^\\s:]+:\\s+$", "match_all": true },
+            { "key": "selection_empty", "match_all": true },
+            { "key": "selector", "operand": "source.yaml.sublime.syntax" },
+            { "key": "setting.packagedev.syntax.input_helpers" },
+        ]
+    },
+    {   // [optional] move caret to the next line and indent by one level.
+        "keys": ["enter"], "command": "insert", "args": {"characters": "\n  "}, "context": [
+            { "key": "preceding_text", "operator": "regex_match", "operand": "^ {4,}-\\s*match:.*$", "match_all": true },
```

Comment:
This definitely looks doable with indentation rules. I'll take a look later.

Reply:
Finally found the Oniguruma Indentation Rules to be the one which is applied after match. So adding separate rules file didn't work.